src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -157,16 +158,22 @@ private TestCase MakeTestCaseFromXmlNode(XmlNode testNode)
         // public for testing
         public DiaNavigationData GetNavigationData(string className, string methodName)
         {
-            if (this.DiaSession == null) return null;
+            if (this.DiaSession == null || TargetAssembly == null) return null;
 
-            var navData = DiaSession.GetNavigationData(className, methodName);
+            var definingType = TargetAssembly.GetType(className);
+            if (definingType == null) return null;
+
+            var method = definingType.GetMethods().Where(o => o.Name == methodName).OrderBy(o => o.GetParameters().Length).FirstOrDefault();
+            if (method == null) return null;
```

Comment:
That was in part due to the fact that I copied the code "as is" from AsyncMethodHelper.GetClassNameForAsyncMethod. If the method lookup here should be different from the one in AsyncMethodHelper.GetClassNameForAsyncMethod, then both copies should be kept and the copy here should have a different(suitable) lookup criteria.

For me selecting a methods with the fewest number of parameters is just an arbitrary choice, so any other criteria would be acceptable. IMHO if an accurate selection criterion is desired, the information node provided by nUnit framework should contain information about the number of parameters in the test case(and even their types for perfect, near-compiler-quality matching criteria).